quickjs-libc.c

```
@@ -1207,6 +1226,62 @@ static JSValue js_std_fdopen(JSContext *ctx, JSValueConst this_val,
 }
 
 #if !defined(__wasi__)
+#if defined(__MINGW64__)
+static JSValue js_std_tmpfile(JSContext *ctx, JSValueConst this_val,
+                              int argc, JSValueConst *argv)
+{
+    JSRuntime *rt = JS_GetRuntime(ctx);
+    JSThreadState *ts = js_get_thread_state(rt);
+    JSTMPFile *s;
+    JSValue obj;
+    char* fn_template, *env_tmp, *fn_buff;
+    int mk_fd, fn_len;
+
+    obj = JS_NewObjectClass(ctx, ts->std_file_class_id);
+    if (JS_IsException(obj))    
+        return JS_EXCEPTION;
+    s = js_mallocz(ctx, sizeof(*s));
+    if (!s) {
+        JS_FreeValue(ctx, obj);
+        return JS_EXCEPTION;
+    }
+    fn_template = "\\qXXXXXXX";
+    env_tmp = getenv("TMP");
+    if (!env_tmp)  
+        env_tmp = getenv("TEMP");
+    if (env_tmp) {
+        fn_len = strlen(env_tmp);
+        fn_buff = js_malloc(ctx, fn_len + 10);
+        memcpy(fn_buff, env_tmp, fn_len);
+        memcpy(fn_buff + fn_len, fn_template, 10);
+    } else {
+        fn_buff = js_mallocz(ctx, 10);
```

You can share the allocation and copy code across both branches by checking if you have an env var or not.

---

It's wrong anyway. Even if there's no %TEMP% I put "/" and that would default to c:\. That defeats the whole point. There are no create permissions there. I should throw an exception if I don't have TEMP. Otherwise I should have added "\\" manually. 
Let me know if you disagree. My original logic was to create in CWD if someone's stripped the environment.

---

There's no way around two ifs that I could see. 
I admit, may be clearer for someone else to read the way you're suggesting.